frontend/packages/dev-console/src/components/topology/list-view/TopologyListView.tsx

```diff
@@ -0,0 +1,221 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import { DataList, Stack, StackItem } from '@patternfly/react-core';
+import { isGraph, Node, isNode, Visualization, GraphElement } from '@patternfly/react-topology';
+import { setQueryArgument, removeQueryArgument } from '@console/internal/components/utils';
+import { TYPE_APPLICATION_GROUP } from '../components';
+import TopologyFilterBar from '../filters/TopologyFilterBar';
+import { getTopologySearchQuery, TOPOLOGY_SEARCH_FILTER_KEY } from '../filters';
+import { TopologyListViewAppGroup } from './TopologyListViewAppGroup';
+import { getChildKinds, sortGroupChildren } from './list-view-utils';
+
+import './TopologyListView.scss';
+import { TopologyListViewUnassignedGroup } from './TopologyListViewUnassignedGroup';
+import { useDeepCompareMemoize } from '@console/shared/src';
+
+export const FILTER_ACTIVE_CLASS = 'odc-m-filter-active';
+
+interface TopologyListViewProps {
+  visualization: Visualization;
+  application: string;
+  namespace: string;
+  sideBar: React.ReactNode;
+  selectedIds: string[];
+  onSelect: (ids: string[]) => void;
+}
+
+const TopologyListView: React.FC<TopologyListViewProps> = ({
+  visualization,
+  sideBar,
+  selectedIds,
+  onSelect,
+}) => {
+  const searchParams = window.location.search;
+  const selectedId = React.useMemo(() => {
+    return selectedIds[0];
+  }, [selectedIds]);
```

Comment:
What am I missing here? This isn't computational... why memoize? The only thing I think it can do is if you mutate the reference of selectIds in the parent you won't change the selectedId here...

Reply:
Correct, there was more here at one point and I just forgot to remove the memo.